src/session_server.c

```diff
@@ -329,6 +329,106 @@ nc_sock_listen_unix(const char *address, const struct nc_server_unix_opts *opts)
     return -1;
 }
 
+/**
+ * @brief Evaluate socket name for AF_UNIX socket.
+ * @param[in] acc_sock_fd is file descriptor for the accepted socket (a nonnegative).
+ * @param[out] host is pointer to char* to which the socket name will be set. It must not be NULL.
+ * @return 0 in case of success. Call free function for parameter host to avoid a memory leak.
+ * @return 0 if the stream socket is unnamed. Parameter host is set to NULL.
+ * @return -1 in case of error. Parameter host is set to NULL.
+ */
+static int
+sock_host_unix(int acc_sock_fd, char **host)
+{
+    char *sun_path;
+    struct sockaddr_storage saddr;
+    socklen_t addr_len;
+
+    *host = NULL;
+    saddr.ss_family = AF_UNIX;
+    addr_len = sizeof(saddr);
+
+    if (getsockname(acc_sock_fd, (struct sockaddr *)&saddr, &addr_len)) {
+        ERR("getsockname failed (%s).", strerror(errno));
+        return -1;
+    }
+
+    sun_path = ((struct sockaddr_un *)&saddr)->sun_path;
+    if (!sun_path) {
+        /* stream socket is unnamed */
+        return 0;
+    }
+
+    if (!(*host = strdup(sun_path))) {
+        ERRMEM;
+        return -1;
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Evaluate socket name and port number for AF_INET socket.
+ * @param[in] saddr is pointer to sockaddr_storage filled by accept function which was successful.
+ * @param[out] host is pointer to char* to which the socket name will be set. It must not be NULL.
+ * @param[out] port is pointer to uint16_t to which the port number will be set. It must not be NULL.
+ * @return 0 in case of success. Call free function for parameter host to avoid a memory leak.
+ * @return -1 in case of error. Parameter host is set to NULL and port is unchanged.
+ */
+static int
+sock_host_inet(const struct sockaddr_storage *saddr, char **host, uint16_t *port)
```

Comment:
The idea was to make `saddr` of type `const struct sockaddr_in *`, the function does not expect anything else.